From 1c61403efad549efa62251d8e4e98c955987f511 Mon Sep 17 00:00:00 2001
From: Jacopo Dall'Aglio <jacopo.dallaglio@kynetics.com>
Date: Thu, 10 Dec 2020 19:09:11 +0000
Subject: [PATCH] fbmem.c: fix formula for centered logo margins

If logo width is equal to (or up to 8 pixels less than) the
number of pixel columns (xres) the horizontal margin calculated
is negative if with the following formula:
(xres - n * (logo->width + 8) - 8) / 2
Use instead:
(xres - n * logo->width) / 2

Signed-off-by: Jacopo Dall'Aglio <jacopo.dallaglio@kynetics.com>
---
 drivers/video/fbdev/core/fbmem.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/core/fbmem.c b/drivers/video/fbdev/core/fbmem.c
index 0ac6fb3..04737d2 100644
--- a/drivers/video/fbdev/core/fbmem.c
+++ b/drivers/video/fbdev/core/fbmem.c
@@ -512,7 +512,7 @@ static int fb_show_logo_line(struct fb_info *info, int rotate,
 
 		while (n && (n * (logo->width + 8) - 8 > xres))
 			--n;
-		image.dx = (xres - n * (logo->width + 8) - 8) / 2;
+		image.dx = (xres - n * logo->width) / 2;
 		image.dy = y ?: (yres - logo->height) / 2;
 	} else {
 		image.dx = 0;
-- 
2.7.4

